getting_started/scripting/cross_language_scripting.rst

```diff
@@ -0,0 +1,213 @@
+GDScript side knows about.
+Essentially, you want to work with built-in types as described in :ref:`doc_gdscript` or classes extending :ref:`class_Object`.
+
+Calling methods
+---------------
+
+Calling C# methods from GDScript
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Again, calling C# methods from GDScript should be straightforward. The
+marshalling process will do its best to cast your the arguments to match
+function signatures.
+If that's impossible you'll see the following error: ``Invalid call. Nonexistent function `FunctionName```.
+
+.. code-block:: gdscript
+
+    my_csharp_node.PrintNodeName(self) # myGDScriptNode
+    # my_csharp_node.PrintNodeName() # This line will fail.
+
+    my_csharp_node.PrintArray(["a", "b", "c"]) # a, b, c
+    my_csharp_node.PrintArray([1, 2, 3]) # 1, 2, 3
+
+Calling GDScript methods from C#
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To call GDScript methods from C# you'll need to use
+:ref:`Object.Call() <class_Object_method_call>`. The first arguments is the
+name of the method you want to call. The following arguments will be passed
+to said method.
+
+.. code-block:: csharp
+
+    myGDScriptNode.Call("print_node_name", this); // my_csharp_node
+    // myGDScriptNode.Call("print_node_name"); // This line will fail silently and won't error out.
+
+    String[] arr = new String[] { "a", "b", "c" };
+    myGDScriptNode.Call("print_array", arr); // This line will fail silently and won't error out.
+    myGDScriptNode.Call("print_array", new object[] { arr }); // a, b, c
+    myGDScriptNode.Call("print_array", new object[] { new int[] { 1, 2, 3 } }); // 1, 2, 3
+
+.. warning::
+
+    As you can see, if the first argument of the called method is an array,
+    you'll need to wrap the arguments in another ``object[]`` array.
+    Otherwise each element of your array will be treated as a single argument
+    and the function signature won't match.
+
+Behind the scene
+----------------
+
+Here are some explanation about how this is working.
+
+For :ref:`Object.call() <class_Object_method_call>`:
+
+-   If the method is ``free`` (which is hard-coded), instantly free the object
+    and return
+-   If the object has a script instance and the instance has the method,
+    call it and return
+-   If the script has the method (then it's a static method), call it and
+    return
+-   If ClassDB has a native method bind with the given name for the given
+    class, call it and return
+-   Otherwise, error as ``Invalid method``
+
+For :ref:`Object.get() <class_Object_method_get>` and
+:ref:`Object.set() <class_Object_method_set>`:
+
+-   If the object has a script instance and the instance has the property,
+    get/set it and return
+-   If ClassDB has a native property bind with the given name for the given
+    class, call the native getter/setter and return
+-   Hard-coded ``script`` get/set
+-   Hard-coded ``__meta__`` get/set
+-   Hard-coded ``_sections_unfolded`` get/set
+-   Otherwise, fallback to _setv/_getv
```

Comment:
I think the "behind the scene" section is unnecessary. Most people reading this won't find it useful. Also it would fit better somewhere else, in a page about `Object` internals.